### PR TITLE
Update Spot goal orientation with Aria CPF z axis

### PR DESCRIPTION
**Pull request by KavitShah1998**

### CODE REVIEW

**Review by zephirefaith — COMMENTED** [review]
Makes sense, I assume you've verified this as well. Can we see if the logic may be simplified through sophus library?

File: aria_data_loaders/aria_data_utils/aria_streamer.py [hunk 1]
```diff
@@ -886,10 +888,23 @@ def main(data_path: str, vrs_name: str, dry_run: bool, verbose: bool):
...
+    cpf_z_axis_in_spotWorld = best_object_spotWorld_T_cpf.matrix()[:3, 2]
+    projected_cpf_z_axis_in_spotWorld_xy = np.multiply(
+        cpf_z_axis_in_spotWorld, np.array([1.0, 1.0, 0.0])
+    )
+    orientation = float(
+        np.arctan2(
+            projected_cpf_z_axis_in_spotWorld_xy[1],
+            projected_cpf_z_axis_in_spotWorld_xy[0],
+        )
+    )  # tan^-1(y/x)
```
[the remark this answers is not in the document]
Comment (KavitShah1998) [reply aria_data_loaders/aria_data_utils/aria_streamer.py:902]: sophuspy.SE3 does not have SO2 object, here is the link for sophuspy SE3 APIs- https://github.com/craigstar/SophusPy/blob/master/sophus/python/se3.h

[the remark this answers is not in the document]
Comment (KavitShah1998) [reply aria_data_loaders/aria_data_utils/aria_streamer.py]: Created an issue for tracking this separately - https://github.com/facebookresearch/spot-sim2real/issues/98

File: aria_data_loaders/aria_data_utils/aria_streamer.py [hunk 2]
```diff
@@ -921,14 +921,30 @@ def main(
...
+            # Get the position and orientation of the object of interest in spotWorld frame
             pose_of_interest = next_object_spotWorld_T_cpf
+
+            # Position is obtained from the translation component of the pose
             position = pose_of_interest.translation()
-            logger.debug(f" Going to {next_object=} at {position=}")
+
+            # Find the angle made by CPF's z axis with spotWorld's x axis
+            # as robot should orient to the CPF's z axis
+            cpf_z_axis_in_spotWorld = pose_of_interest.matrix()[:3, 2]
```
Comment (jimmytyyang) [inline aria_data_loaders/aria_data_utils/aria_streamer.py:933]: Could we add some context for the magic numbers here?

File: aria_data_loaders/aria_data_utils/aria_streamer.py [hunk 3]
```diff
@@ -921,14 +921,30 @@ def main(
...
+            projected_cpf_z_axis_in_spotWorld_xy = np.multiply(
+                cpf_z_axis_in_spotWorld, np.array([1.0, 1.0, 0.0])
```
Comment (jimmytyyang) [inline aria_data_loaders/aria_data_utils/aria_streamer.py:935]: Could we add a variable name for `np.array([1.0, 1.0, 0.0])`?

**Review by jimmytyyang — COMMENTED** [review]
Looks good to me! Just NIT, and please let me know if you have any questions!

So is this .py is an entry point for the demo?

File: aria_data_loaders/aria_data_utils/aria_streamer.py (hunk 2 above)
Comment (KavitShah1998) [reply aria_data_loaders/aria_data_utils/aria_streamer.py:934]: Addressed - 707b6e918d97686e3027aeffb58718d5b54c0799

File: aria_data_loaders/aria_data_utils/aria_streamer.py (hunk 3 above)
Comment (KavitShah1998) [reply aria_data_loaders/aria_data_utils/aria_streamer.py]: Addressed here - 707b6e918d97686e3027aeffb58718d5b54c0799

**Review by rutadesai — APPROVED** [review]
LGTM! Thanks for fixing this :)

**Review by jimmytyyang — APPROVED** [review]
LGTM!!!! THanks you Sir !!!!!

**Pull request merged**